src/net_processing.cpp

```diff
@@ -398,7 +398,7 @@ void UpdateBlockAvailability(NodeId nodeid, const uint256 &hash) {
 void MaybeSetPeerAsAnnouncingHeaderAndIDs(NodeId nodeid, CConnman& connman) {
     AssertLockHeld(cs_main);
     CNodeState* nodestate = State(nodeid);
-    if (!nodestate->fSupportsDesiredCmpctVersion) {
+    if (!nodestate || !nodestate->fSupportsDesiredCmpctVersion) {
```

Comment:
If the peer is disconnected between when it provides the block and when the block is connected it can, sure.